src/Sylius/Bundle/ApiBundle/Validator/Constraints/OrderPaymentMethodEligibilityValidator.php

```diff
@@ -36,14 +37,19 @@ public function validate(mixed $value, Constraint $constraint): void
 
         /** @var OrderInterface|null $order */
         $order = $this->orderRepository->findOneBy(['tokenValue' => $value->getOrderTokenValue()]);
+
         Assert::notNull($order);
 
+        $channel = $order->getChannel();
+        Assert::notNull($channel);
+
         /** @var PaymentInterface $payment */
         foreach ($order->getPayments() as $payment) {
-            if (!$payment->getMethod()->isEnabled()) {
+            $paymentMethod = $payment->getMethod();
+            if ($paymentMethod instanceof PaymentMethodInterface && (!$paymentMethod->isEnabled() || !$paymentMethod->hasChannel($channel))) {
```

Comment:
```suggestion
            /** @var PaymentMethodInterface $paymentMethod */
            $paymentMethod = $payment->getMethod();
            if (!$paymentMethod->isEnabled() || !$paymentMethod->hasChannel($channel)) {
```
We're in API context, it's already at the very least a Core payment method